Added missing Chaincode name for the deploy api

cx0der opened the pull request:
Without the `name` attribute deploy API fails with https://github.com/hyperledger/fabric/blob/95d2b1ac230bc5abffa12929315053a5e1df04fa/core/rest/rest_api.go#L695

cx0der committed:
Added missing Chaincode name for the deploy api

Without the `name` attribute deploy API fails with https://github.com/hyperledger/fabric/blob/95d2b1ac230bc5abffa12929315053a5e1df04fa/core/rest/rest_api.go#L695

cx0der commented:
Signed-off-by: Shiva t.shivakumar@gmail.com

cx0der committed:
Adding missing name attribute with secure context

cx0der committed:
Correcting field ChaincodeID.name to the value that works

muralisrini commented:
@cx0der : the changes look ok. You shouldn't need "-path" in the "deploy" option in "--peer-chaincodedev" mode (I think... I know that to be true in the "cli" mode). Can you check and make that change too please ?

angrbrd commented:
@cx0der 

@muralisrini is correct in that the "path" is not necessary for a deploy when the peer is running in development mode. Please remove the "path" from the payload in the deploy request with and without security enabled. Otherwise, looks good. Thanks for catching my batch replace during the repo move :-].

cx0der committed:
Remove path attribute from the deploy api call

Removed the path attribute from the deploy api payload as it is not required in development mode.

cx0der commented:
Thanks for the comments, I have removed the path attribute. Also corrected the expected responses.

angrbrd commented:
@cx0der looks great. Thanks.

@srderson @binhn , please merge this change.

srderson commented:
LGTM. Thanks
